src/utils/checkTypes.js

```diff
@@ -1,10 +1,12 @@
 import invariant from 'invariant'
 import isPlainObject from './isPlainObject'
 
-function typecheck(type, name, obj) {
+function typecheck(types, name, obj) {
   invariant(
-    typeof obj === type,
-    `${name} must be a ${type}. Instead received a %s.`,
+    Array.isArray(types)
+      ? types.some(t => typeof obj === t)
+      : typeof obj === types,
+    `${name} must be a ${types}. Instead received a %s.`,
```

Comment:
actually I tested it and an array is coerced to a string correctly, eg.
```js
arr = ['function', 'undefined']
str = `then must be a ${arr}`
```
`str` will be `then must be a function, undefined` which is pretty clear but when types is an array I can replace `a` with `one of` so that it becomes `then must be one of function, undefined`, should I do that?